network/client.go

```diff
@@ -0,0 +1,110 @@
+package network
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/dedis/crypto/config"
+)
+
+// Client is an interface which is a simpler version of a Router. The main use
+// for a Client is to directly Send something and get a result back. It is used
+// intensively by Services to have a easy external API.
+// Two implementations are done: TcpClient to use for applications and
+// deployement,and localClient to use for local testing alongside with
+// LocalRouter.
+// NOTE: This interface is likely to be removed to be replaced by a full
+// pledged REST HTTP Api directly connected to the sda/services.
+type Client struct {
+	connector func(own, remote *ServerIdentity) (Conn, error)
+}
+
+func newClient(c func(own, remote *ServerIdentity) (Conn, error)) *Client {
+	return &Client{c}
+}
+
+var baseID uint64
+var baseIDLock sync.Mutex
+
+var timeoutResponse = 10 * time.Second
+
+// Send will send the message to the destination service and return the
+// reply.
+// The error-handling is done using the ErrorRet structure which can be returned
+// in place of the standard reply. This method will catch that and return
+// the appropriate error as a network.Packet.
+// Send will timeout and return an error if it has not received any response
+// under 10 sec.
+func (cl *Client) Send(dst *ServerIdentity, msg Body) (*Packet, error) {
+	kp := config.NewKeyPair(Suite)
+	// just create a random looking id for this client. Choosing higher values
+	// lower the chance of having a collision in the Router.
+	baseIDLock.Lock()
+	id := baseID
+	baseID++
+	baseIDLock.Unlock()
+	sid := NewServerIdentity(kp.Public, NewAddress(dst.Address.ConnType(), "client:"+strconv.FormatUint(id, 10)))
+
+	c, err := cl.connector(sid, dst)
+	if err != nil {
+		return nil, fmt.Errorf("Could not connect %x", err)
+	}
+	defer c.Close()
+
+	if err := c.Send(sid); err != nil {
+		return nil, err
+	}
+
+	msgCh := make(chan Packet)
+	errCh := make(chan error)
+	go func() {
+		if err := c.Send(msg); err != nil {
+			errCh <- err
+			return
+		}
+		p, err := c.Receive()
+		if ret := ErrMsg(&p, err); ret != nil {
+			errCh <- ret
+		} else {
+			msgCh <- p
+		}
+	}()
+
+	select {
+	case resp := <-msgCh:
+		return &resp, nil
+	case err := <-errCh:
+		return nil, err
+	case <-time.After(timeoutResponse):
+		return nil, errors.New("Timeout on sending message")
+	}
+}
+
+// StatusRet is used when a status is returned - mostly an error
+type StatusRet struct {
+	Status string
+}
+
+// StatusOK is used when there is no error but nothing to return
+var StatusOK = &StatusRet{""}
```

Comment:
Not sure but it's a good thing to have in my opinion, if it's possible to register a function using the `ServiceProcessor` using this as a return value. I mean, it will surely come the time when a Service does not need to answer to a client (for example an update or something).